### DISCUSSION TITLE
update resets comp flag

### DISCUSSION THREAD

**Discussion by trapd00r** [beetbox/beets#3990] [Q&A]
I don't know if this is a bug or if I'm doing something wrong, but this happens from time to time. During a `beet update` the comp flag is reset to 0 and the albumartist is modified so that all of a sudden the individual album tracks gets spread all around my library in individual artist directories. Why is this?

```
/mnt/music8/+TAGGED/+VA/This Is Goa Trance Anthems 2 (2012)/23 Va - This Is Goa Trance Anthems 2 Part 2 Continuous Mix.mp3
  comp: True -> False
  albumartist: Various Artists -> Va
  genre: Goa -> Psychedelic
/mnt/music8/+TAGGED/+VA/This Is Goa Trance Anthems 4 (2012)/01 Man With No Name - Big Troubles In Out Space (oforia Remix).mp3
  comp: True -> False
  albumartist: Various Artists -> Man With No Name
  genre: Goa -> Psychedelic
```

**Comment by wisp3rwind** [comment]
Well, without any context, this will be hard to tell. Did you edit the files using some tool other than beets before the `beets update`? If so, are you sure that that tool didn't simply reset the `comp` flag (stored as `TCMP` in mp3s)?

**Reply by trapd00r** [reply]
No, the only thing I've done is importing with beets.

**Reply by wisp3rwind** [reply]
Could there be a media player messing with the tags? The reason I'm suspicious is that the `genre` tag changes. Looking at the the code, I can't immediately exclude that there's an error with our logic for handling `comp` and `albumartist`, but that change of `genre` would be pretty weird without any other program interfering.

It might helpful to see the output of `mutagen-inspect <file>` for such a file before and after the `beet update`.

**Reply by trapd00r** [reply]
1) No; I'm using mpd and it doesn't alter the files in any way that I'm aware of.
2) Cool, didn't know about the suite of mutagen-* tools. The hard thing would be to know, beforehand, when things will be screwed up though. Guess I'll make a habit of supplying the -p flag to update.

**Reply by wisp3rwind** [reply]
>     2. Cool, didn't know about the suite of mutagen-* tools. The hard thing would be to know, beforehand, when things will be screwed up though. Guess I'll make a habit of supplying the -p flag to update.

Yeah, that's certainly inconvenient. When this shows up again, I'd be great if you could provide an example of what kind of tags in the file trigger this issue.
